### PR TITLE
Jordanjanakievski/tces 74 filter by employer company name for job leads dashboard

### PR DESCRIPTION
**Pull request by jordanjanakievski**
# Description

Adds the ability to filter by the company name in job leads. We will also be able to use this query for the job leads component on the employer pages

Issue #74

- [X] New feature (non-breaking change which adds functionality)

# Checklist

- [X] My code follows the style guidelines of this project (required)
- [x] I have performed a self-review of my code (required)
- [x] I have linted and formatted my code (required)
- [x] New and existing unit tests pass locally with my changes (required)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by kevinle623 — DISMISSED** [review]
clean

**Review by DanielDervishi — DISMISSED** [review]
Great Work Jordan 😎

File: server/src/controllers/job_lead/getAllJobLeads.js [hunk 1]
```diff
@@ -54,6 +56,21 @@ const getAllJobLeadsRequestHandler = async (req, res) => {
...
+          name: { [Op.like]: `%${searchEmployerNameQuery}%` },
+        },
+        attributes: ["id"],
+      });
+      employerIds = employers.map((employer) => employer.id);
+    }
+
+    if (employerIds.length > 0) {
+      query.employer = { [Op.in]: employerIds };
+    }
```
Comment (DanielDervishi) [inline server/src/controllers/job_lead/getAllJobLeads.js]: I think we should move this code block inside of the "`if (searchEmployerNameQuery)`" statement. Alongside this change I think we should remove the "`if (employerIds.length > 0)`" check. The result of applying these two changes is that whenever "searchEmployerNameQuery" is set, we only return the employers with the given name.
In the current implementation if no employer matches the given name, we return all of them.

**Review by DanielDervishi — COMMENTED** [review]
Just thought about a potential issue after looking at a different pr. Sorry for the confusion.

File: server/src/controllers/job_lead/getAllJobLeads.js (hunk 1 above)
Comment (jordanjanakievski) [reply server/src/controllers/job_lead/getAllJobLeads.js]: Ok got that fixed up

**Review by DanielDervishi — APPROVED** [review]

**Review by kevinle623 — APPROVED** [review]

**Pull request merged**